src/gsi/setuprad.f90

```diff
@@ -996,6 +999,14 @@ subroutine setuprad(obsLL,odiagLL,lunin,mype,aivals,stats,nchanl,nreal,nobs,&
            varinv(1:nchanl) = zero
         endif
 
+! Include a separate check for NaNs in CRTM calculations.  These are not always caught by other tests
+
+        if (ANY(ieee_is_nan(tsim(1:nchanl)))) then
+           write(*,*) 'WARNING: NaN found in CRTM coefficent output'
```

Comment:
Suggestion:   It would be more informational if we print out the latitude/longitude  (cenlat/cenlon) and channels where the NAN occurs as part of the warning message.